server/src/user/user.service.ts

```diff
@@ -136,6 +139,51 @@ export class UserService {
     }
   }
 
+  async changePassword(userId: number, dto: ChangePasswordDto) {
+    try {
+      const user = await this.prisma.user.findUnique({
+        where: {
+          id: userId,
+        },
+      });
+
+      if (!user) {
+        throw new NotFoundException(`User with ID ${userId} not found.`);
+      }
+
+      const passwordMatch = await compare(
+        dto.currentPassword,
+        user.hashedPassword,
+      );
+
+      if (!passwordMatch) {
+        throw new UnauthorizedException('Invalid credentials');
```

Comment:
I think in this case, it's better to send `The current password is not correct` as the error message!

Reply:
Fixed it.
d3c093a061236221f5429c0eb4b6d5312e98515a